src/wallet/hardwaresigner.rs

```diff
@@ -12,6 +12,62 @@
 //! HWI Signer
 //!
 //! This module contains a simple implementation of a Custom signer for rust-hwi
+//! ```no_run
+//! use bdk::bitcoin::Network;
+//! use bdk::database::MemoryDatabase;
+//! use bdk::signer::SignerOrdering;
+//! use bdk::wallet::hardwaresigner::HWISigner;
+//! use bdk::wallet::AddressIndex::New;
+//! use bdk::{FeeRate, KeychainKind, SignOptions, SyncOptions, Wallet};
+//! use hwi::{types::HWIChain, HWIClient};
+//! use std::sync::Arc;
+//!
+//! fn main() -> Result<(), bdk::Error> {
+//!     // Listing all the available hardware wallet devices...
+//!     let devices = HWIClient::enumerate().unwrap();
+//!     // ...and creating a client out of the first one
+//!     let client = HWIClient::get_client(
+//!         devices.first().expect(
+//!             "No devices found. Either plug in a hardware wallet, or start a simulator.",
+//!         ),
```

Comment:
Consider assigning `devices.first()` to a variable because you use it down below as well